homeassistant/components/guardian/__init__.py

```diff
@@ -410,14 +391,14 @@ def __init__(
         self._attr_unique_id = f"{paired_sensor_uid}_{description.key}"
 
 
-@dataclass(frozen=True)
+@dataclass(frozen=True, kw_only=True)
 class ValveControllerEntityDescriptionMixin:
```

Comment:
Do you need the mixins with the kw_only ?

Reply:
Tell me more. I was using mixins to add additional required args in a composable fashion—does `kw_only` allow me to not do that?

Reply:
Maybe something like bb6f78dcc49a9a6afad56173d5ea01d5ad236b4e 40f30675dd25713a1f25f00159d303874c96c07b

Reply:
Fascinating! What is `kw_only` doing that allows this? Or are they disconnected?

I seem to recall that I used mixins when I had a mixture of perimeters with and without defaults values, but it's been a long time.